app/models/concerns/pre_sale_prices/product_pre_saleable.rb

```diff
@@ -13,8 +13,8 @@ module ProductPreSaleable
       delegate :pre_sale_price, to: :master
     end
 
-    def on_sale?
-      pre_sale_prices.count > 0
+    def on_sale?(currency)
+      pre_sale_prices.in_currency(currency).sum(:amount) > 0
```

Comment:
@nickwood https://github.com/madetech/spree_pre_sale_prices/pull/6/commits/13bc5d098b3dc6f1a9b42e98334aa582362819f0 this happens